x/wasm/keeper/keeper.go

```diff
@@ -640,8 +648,8 @@ func (k Keeper) reply(ctx sdk.Context, contractAddress sdk.AccAddress, reply was
 		return nil, err
 	}
 
-	// always consider this pinned
-	replyCosts := k.gasRegister.ReplyCosts(true, reply)
+	ctx, discount := k.checkDiscountEligibility(ctx, string(codeInfo.CodeHash), k.IsPinnedCode(ctx, contractInfo.CodeID))
+	replyCosts := k.gasRegister.ReplyCosts(discount, reply)
```

Comment:
discount should always be `true` here, right? I think the new implementation consumes more gas because the `IsPinnedCode` call checks the store.

Reply:
agreed, here we can just say `discount := true`